libtailscale/localapi.go

```diff
@@ -103,6 +108,12 @@ func (app *App) CallLocalAPIMultipart(timeoutMillis int, method, endpoint string
 	}
 }
 
+func (app *App) EditPrefs(prefs ipn.MaskedPrefs) (LocalAPIResponse, error) {
+	r, w := io.Pipe()
```

Comment:
You didn't write the masked prefs to the pipe here so the patch body will be empty.